src/modules/common/components/bars/topBar/networkName.js

```diff
@@ -1,42 +1,45 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Dropdown from 'src/theme/Dropdown/dropdown';
+import Icon from 'src/theme/Icon';
 
+import { networkSelected } from 'src/redux/actions';
 import { networkKeys } from '@network/configuration/networks';
 import { getNetworkName } from '@network/utils/getNetwork';
-import Tooltip from 'src/theme/Tooltip';
 import styles from './network.css';
 
-const Network = ({ network, t, token }) => {
-  const networksList = {
-    [networkKeys.mainNet]: t('Mainnet').toLowerCase(),
-    [networkKeys.testNet]: t('Testnet').toLowerCase(),
-    [networkKeys.customNode]: t('Custom node').toLowerCase(),
-  };
+const Network = ({ t }) => {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const network = useSelector(state => state.network);
+  const dispatch = useDispatch()
   const activeNetwork = getNetworkName(network);
+
   const statusColor = network.status.online ? styles.online : styles.offline;
 
+  const closeDropdown = () => {
+    setShowDropdown(false)
+  }
+
+  const setActiveNetwork = net => {
+    dispatch(networkSelected({ name: net }))
+    closeDropdown();
+  }
+
+  const activeNetworkIndex = useMemo(() => Object.values(networkKeys).indexOf(activeNetwork), [activeNetwork])
+
   return (
-    <section className={styles.wrapper}>
-      <span className={`${styles.status} ${statusColor}`} />
-      <div className={styles.message}>
-        <span>{t('Connected to:')}</span>
-        <Tooltip
-          className={styles.tooltipWrapper}
-          size="maxContent"
-          position="bottom left"
-          content={(
-            <span className="network-name">{networksList[activeNetwork]}</span>
-          )}
-        >
-          <p className="network-address">
-            {
-              network.networks
-                ? network.networks[token]?.serviceUrl
-                : '-'
-            }
-          </p>
-        </Tooltip>
-      </div>
-    </section>
+    <>
+      <section className={styles.wrapper} onClick={() => setShowDropdown(true)} >
+        <span className={`${styles.status} ${statusColor}`} />
+        <div className={styles.message}>
+          <span >{activeNetwork}</span>
```

Comment:
@Balanced02 Please remove space before span closing tag (`<span >`) and other similar cases (` >`)